addons/l10n_ec/models/res_partner.py

```diff
@@ -49,9 +49,24 @@ def check_vat(self):
                         error_message = ""
                         if partner.l10n_latam_identification_type_id.id == it_dni.id:
                             error_message = _("VAT %s is not valid for an Ecuadorian DNI, "
-                                              "it must be like this form 0915068258") % partner.vat
+                                              "it must be like this form 1234567897") % partner.vat
                         if partner.l10n_latam_identification_type_id.id == it_ruc.id:
                             error_message = _("VAT %s is not valid for an Ecuadorian company, "
-                                              "it must be like this form 0993143790001") % partner.vat
+                                              "it must be like this form 1234567897001") % partner.vat
                         raise ValidationError(error_message)
         return super(ResPartner, self - ecuadorian_partners).check_vat()
+
+    def _l10n_ec_get_identification_type(self):
+        # Maps Odoo identification types to Ecuadorian ones, useful for document type domains, electronic documents, ats, others
+        idtype = False
+        if self.l10n_latam_identification_type_id == self.env.ref('l10n_ec.ec_dni'):
+            idtype = 'cedula' #dni
+        elif self.l10n_latam_identification_type_id == self.env.ref('l10n_ec.ec_ruc'):
+            idtype = 'ruc' # RUC
+        elif self.l10n_latam_identification_type_id == self.env.ref('l10n_latam_base.it_pass'):
+            idtype = 'passport'  # Pasaporte
+        elif self.l10n_latam_identification_type_id == self.env.ref('l10n_latam_base.it_fid', 'l10n_latam_base.it_vat'):
+            idtype = 'foreign' # Identificacion del exterior
+        elif self.l10n_latam_identification_type_id.country_id != self.env.ref('base.ec'):
+            idtype = 'foreign' # Identificacion del exterior
+        return idtype
```

Comment:
```suggestion
    def _l10n_ec_get_identification_type(self):
        # Maps Odoo identification types to Ecuadorian ones, useful for document type domains, electronic documents, ats, others
        self.ensure_one()
        if self.l10n_latam_identification_type_id == self.env.ref('l10n_ec.ec_dni'):
            return 'cedula' #dni
        elif self.l10n_latam_identification_type_id == self.env.ref('l10n_ec.ec_ruc'):
            return 'ruc' # RUC
        elif self.l10n_latam_identification_type_id == self.env.ref('l10n_latam_base.it_pass'):
            return 'passport'  # Pasaporte
        elif self.l10n_latam_identification_type_id == self.env.ref('l10n_latam_base.it_fid', 'l10n_latam_base.it_vat'):
            return 'foreign' # Identificacion del exterior
        elif self.l10n_latam_identification_type_id.country_id != self.env.ref('base.ec'):
            return 'foreign' # Identificacion del exterior
```
plz add a proper docstring instead of a random comment at the top

Reply:
According to the docs, `self.env.ref` only takes one identifier. Tha other parameter is probably being converted to a boolean as the `raise_if_not_found` parameter. That's probably not intended. I would rewrite the 'foreign' option as a single condition.

Reply:
@odoostan Better to do country_id.code != 'EC' here, but I guess it is old code based.  I am also wary of self.l10n_latam_identification_type_id == self.env.ref('l10n_latam_base.it_pass')

`(Pdb) self.env['account.move'].browse(59).partner_id.l10n_latam_identification_type_id == self.env.ref('l10n_latam_base.it_pass', 'l10n_ec.ec_dni')
False
(Pdb) self.env['account.move'].browse(59).partner_id.l10n_latam_identification_type_id
l10n_latam.identification.type(5,)
(Pdb) self.env.ref('l10n_ec.ec_dni')
l10n_latam.identification.type(5,)
`

Reply:
@jco-odoo let me know what you think of my latest commit. Could be easily written with a dict as well :) could make the local function simpler and just call it twice for the foreign case.